pkg/util/azureclient/mgmt/resources/deployments_addons.go

```diff
@@ -22,8 +26,15 @@ func (c *deploymentsClient) CreateOrUpdateAndWait(ctx context.Context, resourceG
 	if err != nil {
 		return err
 	}
-
-	return future.WaitForCompletionRef(ctx, c.Client)
+	// all error codes: https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors
+	err = future.WaitForCompletionRef(ctx, c.Client)
+	if err != nil {
```

Comment:
im
working on the assumption it will be this error format
```
{
  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "BadRequest",
      "message": "{\r\n  \"error\": {\r\n    \"code\": \"InUseSubnetCannotBeDeleted\",\r\n    \"message\": \"Subnet ljakubow-v4test-master is in use by /subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourceGroups/aro-cay8344w/providers/Microsoft.Network/loadBalancers/aro-internal-lb/frontendIPConfigurations/internal-lb-ip and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet.\",\r\n    \"details\": []\r\n  }\r\n}"
    }
  ]
}
```

And from the code Resp.Code = StatusCode. Not sure how to "re-parse" error message.

Reply:
im little bit lost on this implementation tbh

Reply:
in golang, construct a deployment that will fail due to quota errors and use spew.Dump to see the details of the error you get back.  That'll tell you how to parse it.